lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -1884,6 +1906,46 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     visitPart(RST->getReferentType(), optionalKind);
   }
   
+  Optional<ValueDecl *> findRenamedDecl(const swift::AvailableAttr *AvAttr, const swift::Decl *D) {
+    if (!isa<ValueDecl>(D))
+      return None;
+    
+    SmallVector<ValueDecl *, 4> lookupResults;
+    auto renamedParsedDeclName = swift::parseDeclName(AvAttr->Rename);
+    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
+    
+    auto declContext = D->getDeclContext();
+    Optional<ValueDecl *>renamedFuncDecl = None;
+
+    if (isa<ClassDecl>(D) || isa<ProtocolDecl>(D)) {
+      UnqualifiedLookup lookup(renamedDeclName.getBaseIdentifier(), declContext->getModuleScopeContext(), nullptr);
+      renamedFuncDecl = lookup.getSingleTypeResult();
+    } else {
+      declContext->lookupQualified(declContext->getSelfTypeInContext(),
+                                   renamedDeclName, NL_QualifiedDefault, NULL,
+                                   lookupResults);
+      for (auto candidate : lookupResults) {
+        if (!shouldInclude(candidate))
+          continue;
+        
+        if (candidate->getKind() != D->getKind())
+          continue;
+        
+        if (candidate->getKind() == DeclKind::Func && isa<FuncDecl>(candidate)
+            && cast<FuncDecl>(candidate)->getParameterLists().size() !=
+            cast<FuncDecl>(D)->getParameterLists().size())
+          continue;
```

Comment:
I think you're trying to check that the number of parameters is the same here? That's not a bad idea, but it's not `getParameterLists()` that you want. Fortunately Slava P just added a useful API for what you do want: `getParameters()`.

Another good thing to check is whether they're both instance members.

Reply:
THAT IS GREAT!!! Thank you Slava P :D